File: newton_raphson.py
#  Newton Raphson Method for solving equations
import sympy as sym


def func( x ):
    
	return x * x * x - x * x + 2

# Derivative of the above function
# which is 3*x^x - 2*x
def derivFunc( x ):
	return 3 * x * x - 2 * x

# Function to find the root
def newtonRaphson( x ):
	h = func(x) / derivFunc(x)
	while abs(h) >= 0.0001:
		h = func(x)/derivFunc(x)
		
		# x(i+1) = x(i) - f(x) / f'(x)
		x = x - h
	
	print("The value of the root is : ",
							"%.4f"% x)

# Driver program to test above
x0 = 1 # Initial values assumed
newtonRaphson(x0)

# x = sym.Symbol('x')
# function = x ** 4 + 7 * x ** 3 + 8
# derivitive = sym.diff(function)
# print(derivitive)

File: newton_raphson.m
% % function { } = newton_raphson( fun , diff, x0)
    
% % x = x0;
% % maxiter = 200;
% % for i = 1:maxiter
% %     x(i+1) = x(i) - fun(x(i))/diff(x(i))
% %     if abs (x(i+1) - x(i)) < tol 
% %         fprintf('the root has converged at x = %.10f\n' , x(i+1));
% %     end
% % end


% % Newton-Raphson Algorithm            
%  % Find the root of y=cos(x) from o to pi.

%  f = @(x) (cos(x));
%  fd = @(x) (-sin(x));
%  p0 = input('Enter initial approximaation: ');
%  n = input('Enter no. of iterations, n: ');
%  tol = input('Enter tolerance, tol:  ');

%  i = 1;
%  while i <= n
%     d=f(p0)/fd(p0);
%     p0 = p0 - d; 
%     if abs(d) < tol
%        fprintf('\nApproximate solution xn= %11.8f \n\n',p0);
%        break;
%     else
%        i = i+1;
%     end
%  end



























